submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/AccessTagProcessor.kt

```diff
@@ -1,14 +1,15 @@
 package ac.uk.ebi.biostd.submission.processors
 
 import ebi.ac.uk.model.ExtendedSubmission
+import ebi.ac.uk.model.extensions.releaseDate
 import ebi.ac.uk.persistence.PersistenceContext
 
 /**
  * Add all parent submission tags to submitted submission.
  */
 class AccessTagProcessor : SubmissionProcessor {
-
     override fun process(submission: ExtendedSubmission, context: PersistenceContext) {
         submission.accessTags.addAll(context.getParentAccessTags(submission))
+        submission.accessTags.removeIf { it == "Public" && submission.releaseDate.isNullOrBlank().not() }
```

Comment:
I think instead we should filter the tags that we add from parent and add a public tag if released, final results is exactly the same but in this way is more consistent with the real business cases scenarios.

Reply:
Done